### PR TITLE
Feature 538: No lingering weekplans when editing

### LINKED ISSUE #538: 'Overståede Uger' does not update correctly
**Describe the bug**
After editing the last weekplan from the 'Overståede Uger' section to become a future week instead and the 'Overståede Uger' section becomes empty, a spinning circle is displayed as if it is trying to load a weekplan.

**To Reproduce**
1. Create a weekplan with a year less than the current year
2. Make sure this is the only weekplan in 'Overståede Uger'
3. Edit the weekplan with a year greater than the current year

**Expected behavior**
No spinning circle

**Actual behavior**
A grey box with a spinning circle.

**Screenshots**
image

**Environment (please complete the following information):**
- OS: [Android]
- Emulator - Yes
- Nexus 9
- APK Version - 29

**Additional context**

### PR DESCRIPTION
**Pull request by AdrianPlesner**
Fixes #538 
The bug is "fixed" by popping and pushing the weekplan selector screen when editing a weekplan while there is only one weekplan left in either current or previous weekplans, as this is the only time when the bug occurs. 
While this is a bit of a hack, it is the only possible solution we could find. It is a very weird bug, as it follows the same line of execution as deleting the weekplan and making a new one with the new changes, which works fine when done manually in succession.

### CODE REVIEW

**Comment by Determinatoro** [comment]
Please use the following checklist, @social2dk. Feel free to check it off as you go along.

**Code Design** 

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [x] Has all the added code been properly commented

Only check one of the items below:

- [x] The functionality does **not** need to be covered by tests
- [ ] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [ ] All functionality is covered by tests
- [ ] The tests has sensible names
- [ ] By reading the tests you know what they cover
- [ ] The tests cover sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, open the app, and test:
- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

**Comment by Determinatoro** [comment]
Please use the following checklist, @Emil7370. Feel free to check it off as you go along.

**Code Design** 

- [x] The code is in the right place? (Both in terms of folder structure and class structure)
- [x] The code is not over-engineered. Examples of over-engineering: Implemented behavior for future problems     
- [x] This code could not, to the best of my knowledge, have reused existing code
- [x] The code does not introduce functionality that is unnecessary for solving the problem

**Code Readability** 

- [x] Names are meaningful and self-documenting
- [x] It is understandable, by reading the code, what it does
- [x] The code is simple and readable. i.e. it contains no \"hack\", or obscure solution   

**Code Maintainability**

- [x] Has all the added code been properly commented

Only check one of the items below:

- [x] The functionality does **not** need to be covered by tests
- [ ] The functionality needs to be covered by tests

Only check these if the functionality needs to be covered by tests:

- [ ] All functionality is covered by tests
- [ ] The tests has sensible names
- [ ] By reading the tests you know what they cover
- [ ] The tests cover sensible cases. Both happy paths and exception paths
- [ ] The tests cover the full functionality. i.e., The tests fails if some of the requested functionality is missing

**Functionality**

For this part, open the app, and test:
- [x] The features that the code claims to implement, are actually implemented
- [x] The code does not look like it has bugs
- [x] All the new screens are reachable through in-app navigation

**Review by Emil7370 — DISMISSED** [review]

**Comment by social2dk** [comment]
image
The issue still happens to me when taking a future weekplan and editing it to make it a previous weekplan, i.e create a weekplan with year 2022, and change it to 2002.

**Comment by AdrianPlesner** [comment]
> The issue still happens to me when taking a future weekplan and editing it to make it a previous weekplan, i.e create a weekplan with year 2022, and change it to 2002.

Should be fixed now

File: ios/Flutter/.last_build_id [hunk 1]
```diff
@@ -0,0 +1 @@
+67c830bc64f83abcd5ee06eb8b9aea75
```
Comment (social2dk) [inline ios/Flutter/.last_build_id:1]: What is this?

**Review by social2dk — APPROVED** [review]

File: ios/Flutter/.last_build_id (hunk 1 above)
Comment (AdrianPlesner) [reply ios/Flutter/.last_build_id:1]: Jahhhhh

**Review by Emil7370 — APPROVED** [review]

**Pull request merged**